### ISSUE TITLE
Failed to upload crash log

### ISSUE THREAD

**Issue by SummerWu** [firebase/firebase-ios-sdk#12739]
### Description

The app triggered a crash, and then I looked at the firebase background and didn't collect a crash log.

After I set up the vpn on my device, I repeated the above steps and found that the firebase background could collect crash logs.

### Reproducing the issue

Before upgrading to 10.3.0, crash logs can be collected normally, so I am wondering, what changes did 10.3.0 make that affect crash log reporting?

What should I do now to solve this problem？

Please help with the answer.

### Firebase SDK Version

10.3.0

### Xcode Version

15.1

### Installation Method

CocoaPods

### Firebase Product(s)

Crashlytics

### Targeted Platforms

iOS

### If using Swift Package Manager, the project's Package.resolved

<details>
<summary>Expand <code>Package.resolved</code> snippet</summary>
<br>

```json

Replace this line with the contents of your Package.resolved.

```

</details>

### If using CocoaPods, the project's Podfile.lock

<details>
<summary>Expand <code>Podfile.lock</code> snippet</summary>
<br>

```yml

Replace this line with the contents of your Podfile.lock!

```

</details>

**Comment by rizafran** [comment]
Thanks for reporting, @SummerWu. Based on the [release notes](https://firebase.google.com/support/release-notes/ios#version_1030_-_december_5_2022), it seems like there were no changes made in Crashlytics SDK. Could you try updating your SDK to a later version as this contains the latest bug fixes? You may also check our troubleshooting page [here](https://firebase.google.com/docs/crashlytics/troubleshooting?platform=ios).

**Comment by SummerWu** [comment]
> Thanks for reporting, @SummerWu. Based on the [release notes](https://firebase.google.com/support/release-notes/ios#version_1030_-_december_5_2022), it seems like there were no changes made in Crashlytics SDK. Could you try updating your SDK to a later version as this contains the latest bug fixes? You may also check our troubleshooting page [here](https://firebase.google.com/docs/crashlytics/troubleshooting?platform=ios).

I tried to update to the latest version, but encountered compilation issues where swift files could not be found, reporting errors such as:
/FIRCLSRolloutsPersistenceManager.m:25:9: fatal error: 'FirebaseCrashlytics/FirebaseCrashlytics-Swift.h' file not found

In version 10.3.0, I did some debugging and found that an interface request has been timed out. The request is called firebaselogging-pa.googleapis.com. Does this request affect the collection of flash logs?

**Comment by rizafran** [comment]
@SummerWu, it looks like _'FirebaseCrashlytics/FirebaseCrashlytics-Swift.h' file not found_ error has already been fixed in [v10.24.0](https://firebase.google.com/support/release-notes/ios#crashlytics).

**Comment by SummerWu** [comment]
> @SummerWu, it looks like _'FirebaseCrashlytics/FirebaseCrashlytics-Swift.h' file not found_ error has already been fixed in [v10.24.0](https://firebase.google.com/support/release-notes/ios#crashlytics).

I will try to update to v10.24.0, thank you so much
